sklearn/base.py

```diff
@@ -1087,3 +1088,36 @@ def is_outlier_detector(estimator):
         True if estimator is an outlier detector and False otherwise.
     """
     return getattr(estimator, "_estimator_type", None) == "outlier_detector"
+
+
+def _fit_context(*, skip_nested_validation=True):
+    """Decorator to run the fit methods of estimators within a context manager.
```

Comment:
Let's add the added version just in case that this thing become officially public.


```suggestion
    """Decorator to run the fit methods of estimators within a context manager.

       .. versionadded:: 1.3
```

Reply:
It might be worth to give more context on which context manager is used and what it is useful for.

Reply:
My goal was to make this decorator generic and might eventually involve other context managers in the future. For instance a ``try except`` around fit to make an interruptable fit.

Reply:
OK I see.